storage/watchable_store.go

```diff
@@ -320,12 +330,18 @@ func (s *watchableStore) notify(rev int64, evs []storagepb.Event) {
 		if eb.revs != 1 {
 			panic("unexpected multiple revisions in notification")
 		}
+		w.mu.Lock()
```

Comment:
It could be updated from either in synced or unsynced, so I don't think it will be raced?
Do you mean the case when the channel is blocking?